dokka-integration-tests/gradle/src/testExampleProjects/kotlin/ExampleProjectsTest.kt

```diff
@@ -0,0 +1,344 @@
+            format = "html",
+        )
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(TestCaseProvider::class)
+    fun `test Javadoc output`(testCase: TestCase) {
+        assumeTrue(testCase.outputsJavadoc)
+
+        testDokkaOutput(
+            testCase = testCase,
+            format = "javadoc",
+        )
+    }
+
+    private fun testDokkaOutput(
+        testCase: TestCase,
+        format: String,
+    ) {
+        val expectedDataDir = testCase.expectedDataDir.resolve(format)
+        val dokkaOutputDir = testCase.dokkaOutputDir.resolve(format)
+
+        assert(expectedDataDir.exists() && expectedDataDir.isDirectory()) {
+            "Missing expectedDataDir: ${expectedDataDir.toUri()}"
+        }
+
+        testCase.project.runner
+            .addArguments(
+                testCase.dokkaGenerateTask,
+                "--stacktrace",
+            )
+            .build {
+                dokkaOutputDir.shouldBeADirectory()
+
+                withClue({
+                    """
+                    expectedDataDir: ${expectedDataDir.toUri()}
+                    actualOutputDir: ${dokkaOutputDir.toUri()}
+                    """.trimIndent()
+                }) {
+                    withClue("expect file trees are the same") {
+                        val expectedFileTree = expectedDataDir.toTreeString()
+                        val actualFileTree = dokkaOutputDir.toTreeString()
+                        expectedFileTree shouldBe actualFileTree
+                    }
+
+                    withClue("expect directories are the same") {
+                        dokkaOutputDir.shouldHaveSameStructureAs(expectedDataDir, skipEmptyDirs = true)
+                        dokkaOutputDir.shouldHaveSameStructureAndContentAs(expectedDataDir, skipEmptyDirs = true)
+                    }
+                }
+            }
+    }
+
+
+    @ParameterizedTest
+    @ArgumentsSource(TestCaseProvider::class)
+    fun `test build cache`(
+        testCase: TestCase,
+        @TempDir tempDir: Path,
+    ) {
+
+        val buildCacheDir = tempDir.resolve("build-cache").apply {
+            deleteRecursively()
+            createDirectories()
+        }
+
+        testCase.project.settingsGradleKts += """
+            |buildCache {
+            |    local {
+            |        directory = File("${buildCacheDir.invariantSeparatorsPathString}")
+            |    }
+            |}
+        """.trimMargin()
+
+        // Initial build, to populate the build cache.
+        testCase.project.runner
+            .addArguments(
+                testCase.dokkaGenerateTask,
+                "--stacktrace",
+            )
+            .build {
+                output shouldContain "BUILD SUCCESSFUL"
+            }
+
+        // Re-run generate, and verify that the Dokka tasks are cached.
+        testCase.project.runner
+            .addArguments(
+                testCase.dokkaGenerateTask,
+                "--stacktrace",
+                "--build-cache",
```

Comment:
I think this test passes even if we remove `--build-cache`. Shouldn't this test do `clean` before rebuilding to make sure build cache entries are being reused? I would also expect this test to check that tasks are fetch `FROM-CACHE`, rather than them being `UP-TO-DATE`

Reply:
Good catch, fixed, thanks!